File: CleanJava/app/src/debug/java/com/vascome/fogtail/di/ui/main/DeveloperSettingsComponent.java
package com.vascome.fogtail.di.ui.main;

import android.support.annotation.NonNull;

import com.vascome.fogtail.ui.dev_settings.fragments.DeveloperSettingsFragment;

import dagger.Subcomponent;

/**
 * Created by vasilypopov on 11/23/17
 * Copyright (c) 2017 MVPJava. All rights reserved.
 */

@Subcomponent
public interface DeveloperSettingsComponent {

    @Subcomponent.Builder
    interface Builder {
        DeveloperSettingsComponent build();
    }

    void inject(@NonNull DeveloperSettingsFragment developerSettingsFragment);
}

File: MosbyMVPKotlin/app/src/release/java/com/vascome/fogtail/di/ui/main/DeveloperSettingsComponent.kt
package com.vascome.fogtail.di.presentation.main

import dagger.Subcomponent

/**
 * Created by vasilypopov on 11/23/17
 * Copyright (c) 2017 MVPJava. All rights reserved.
 */

@Subcomponent
interface DeveloperSettingsComponent {
    // Hides details of developer settings in the release build type.

    @Subcomponent.Builder
    interface Builder {
        fun build(): DeveloperSettingsComponent
    }
}
